gradle-palantir-java-format/src/main/java/com/palantir/javaformat/gradle/spotless/FormatJavaWorkAction.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.javaformat.gradle.spotless;
+
+import com.google.common.base.Suppliers;
+import com.palantir.javaformat.java.FormatterService;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.util.ServiceLoader;
+import java.util.function.Supplier;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.workers.WorkAction;
+
+public abstract class FormatJavaWorkAction implements WorkAction<FormatJavaParameters> {
+
+    private static final Logger logger = Logging.getLogger(FormatJavaWorkAction.class);
+
+    private static final Supplier<FormatterService> formatterService = Suppliers.memoize(() -> {
+        logger.info("Loading FormatterService in worker daemon");
```

Comment:
Running this locally, does it log this per file or just once per gradle invocation? Not sure is gradle is doing something funky with classloaders even in process isolation that means this may be new each time.